test/test_bagitutils.py

```diff
@@ -4,10 +4,43 @@
 import os
 import zipfile
 from bagitutils import BagHandler
+import tempfile
+import shutil
+import csv
 
 
 class TestBagHandlerMethods(unittest.TestCase):
 
+    def setUp(self):
+        # Create a list of lists representing the output of the
+        # bag.write_csv_files method of the fc_mock.tsv file residing in
+        # the test folder.
+        self.bag_tsv_file = [
+            ['entity:sample_id', 'participant_id', 'file_dos_uri1',
+             'file_dos_uri2', 'file_dos_uri3', 'file_path1', 'file_path2',
+             'file_path3', 'file_type1', 'file_type2', 'file_type3', 'gs_url1',
+             'gs_url2', 'gs_url3', 's3_url1', 's3_url2', 's3_url3',
+             'specimen_uuid', 'upload_file_id1', 'upload_file_id2',
+             'upload_file_id3'],
+             ['s1', 'd1', '', '', '', '', '', '', 'crai', 'cram', '',
+              'gs://junk.crai', 'gs://junk.cram', '', '', '', '', 'sp1', '', '',
+              ''],
+             ['s2', 'd1', '', '', '', '', '', '', 'crai', 'cram', '', '',  '',
+               '', 's3://junk.crai', 's3://junk.cram', '', 'sp1', '', '', ''],
+             ['s3', 'd2', '', '', '', '', '', '', 'crai', 'cram', '',
+              'gs://junk.crai', 'gs://junk.cram', '', '', '', '', 'sp1', '', '',
+               ''],
+             ['s4', 'd3', '', '', '', '', '', '', 'crai', 'cram', '',
+              'gs://junk.crai', 'gs://junk.cram', '', '', '', '', 'sp1', '', '',
+               ''],
+             ['s5', 'd4', '', '', '', '', '', '', 'crai', 'cram', 'bam',
+              'gs://junk.crai', 'gs://junk.cram', 'gs://junk.bam', '', '', '',
+               'sp2', '', '', ''],
+             ['s6', 'd5', '', '', '', '', '', '', 'crai', '', '',
+              'gs://junk.crai', '', '', '', '', '', 'sp3', '', '', '']
+        ]
+
+
```

Comment:
I would have just created and committed the expected file and compared the generated file with it. The end result is a file and that's what gets sent to FC, so that would be the best test. Here you are creating a Python representation of the file, so you run the risk of making an error in "translating". Plus if you need to update this in the future, e.g., add another field, you need to edit the Python code without making a mistake, which would be harder than editing the TSV, IMHO.